src/ws/messageTypes.ts

```diff
@@ -163,7 +177,16 @@ export function validateMessageBody(messageType: MessageType, body: unknown): vo
       throw new Error(`Invalid body for message type ${MessageType.PROTOCOL_ERROR}`);
     }
   } else if (messageType === MessageType.CONNECTION_EVENT) {
-    // lots of other possible things to test, too many to list here for now.
+    const connEventBody = body as ConnectionEventBody;
+    // Ensure body.action is a valid ConnectionEventAction
+    if (!Object.values(ConnectionEventAction).includes(connEventBody.action)) {
```

Comment:
Alas, `JSON.parse()` won't enforce proving that a received string actually belongs to an enumeration. We ... probably assume too much in many other places as well.